### ISSUE TITLE
Support for multiple video elements in window

### ISSUE THREAD

**Issue by skyler91** [relgin/cardspotter#6]
Due to social distancing guidelines, many players have started using webcams along with video conferencing software to play paper magic online (see https://www.reddit.com/r/PlayEDH/). With a decent camera setup, this plugin does a great job of detecting cards but only on the first video element of the page. It would be incredibly useful to allow the plugin to detect cards in any video element within the window!

**Comment by relgin** [comment]
On it! (well tonight after work but I'm happy to help out)

I know someone has already made this work a while back so first I'll try to
see if they can share.

Once there is a way to make it work I would very much appreciate if you
could share how to set it up on the wiki (or with me and I can post it if
you prefer).

No promises but it should hopefully have an early version to share with you
tomorrow evening.

Cheers
Jonas

**Comment by skyler91** [comment]
Thanks for the quick response! I was looking at making changes myself, but if it's already been done, that will make things much easier. Once I get a chance to try it out, I'd be happy to write up instructions to share with the rest of the MTG community.

**Comment by relgin** [comment]
Looks like multi-video for the extension itself will need some work I'm afraid. I created https://github.com/relgin/cardspotter/tree/multi-video for this.

I'll start by adding support for an id tag c++ side (which will split it into multiple matchers), once that is done I'll move on to the JS bit so that it can attach to multiple videos. My JS-fu is not all that so if you want to start tackling that I'm happy for any help.

**Comment by skyler91** [comment]
Unfortunately, I'm not well-versed in JS either, but from an initial look, it may take a bit of refactoring since the existing JS code uses a lot of global variables. I probably won't have time to dig deeper until this weekend at earliest.

**Comment by relgin** [comment]
No worries, I'll update here once I get to the js part. Given that global
js mess it will definitely take me more than tomorrow evening though =/

**Comment by relgin** [comment]
Pushed what I think should be enough on the c++ side (https://github.com/relgin/cardspotter/commit/c2214527474543eb9a944057b4296eeeac95e816) done for the day and will move on to the js tomorrow.

**Comment by relgin** [comment]
long story short I had some issues upgrading to latest emscripten and opencv almost back on track now but not quite, just wanted to post some progress and that I haven't given up on this.

**Comment by relgin** [comment]
Two weeks is not two days, but you should be able to grab latest https://github.com/relgin/cardspotter/tree/multi-video and have the auto detection work. Only the first video has click support but the autodetection should provide an example of how to do it should you be willing. I took the long road via the embind branch but the latest version now has much better encasusation etc etc. Hopefully you can still find some value in this.

**Comment by relgin** [comment]
as of 3.1 multivideo support has been added (as a setting) closing.
